### ISSUE TITLE
Style/FormatStringToken mark for links

### ISSUE THREAD

**Issue by ShockwaveNN** [bbatsov/rubocop#5242]
## Steps to reproduce the problem

1. Create `test.rb` file with content
```
p 'https://ru.wikipedia.org/wiki/%D0%90_(%D0%BA%D0%B8%D1%80%D0%B8%D0%BB%D0%BB%D0%B8%D1%86%D0%B0)'
```
2. Run `rubocop test.rb`

## Result:
Several messages like
```
tmp.rb:1:45: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
p 'https://ru.wikipedia.org/wiki/%D0%90_(%D0%BA%D0%B8%D1%80%D0%B8%D0%BB%D0%BB%D0%B8%D1%86%D0%B0)'
```
This string have nothing to do with strings tokens. Just a link

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.52.0 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-linux)
```

**Comment by Drenmi** [comment]
The false positive occurs on `%B` which is a token for a binary number in a Ruby format string. It might be possible to improve the regular expression of this cop to check whether the string is a URL, which may contain URL encoded characters.

**Comment by matthewswann** [comment]
I'm seeing something that might be a similar issue within date-time formats.

`C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
  LOGGING_DATETIME_FORMAT = '%m-%d-%Y %H:%M:%S.%N'`

**Comment by jeremywadsack** [comment]
@matthewswann see #5223 and #5230

**Comment by dennisvandehoef** [comment]
it also happens if you create a HAML-string as we mock in one of our specs:

```ruby
  let(:template) do
    "render_widget :widget_1\n\
     render_widget :widget_2, position: 'sonstwo'\n\
     fasdfdsfsa fdjksdljfasdf\n\
     %div= render_widget :widget_1,\n :show, position: :testposition\n\
     render_widget :widget_3\n
     render_widget :widget_4,
       position: 'umbruch'\n"
  end
```
